disallow administrator login with default credentials in production e…

s3tezsky opened the pull request:
| Q             | A
| ------------- | ---
|Description, reason for the PR| Disallow administrator login with default credentials in production environment. Login will fail and flash message is thrown.
|New feature| Yes
|[BC breaks](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md)| No
|Fixes issues| closes #489 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

henzigo commented:
Please do not do that 😢 On CI server there is in 99% production envirovment.

LukasHeinz commented:
Don't worry @henzigo , there will be a way how to make it work on CI.

DavidOstrozlik commented on docs/upgrade/UPGRADE-v8.1.0-dev.md:
````diff
@@ -24,6 +24,40 @@ There you can find links to upgrade notes for other versions too.
+            provider: administrators
+            logout_on_user_change: true
+        ```
+        - in case you need to disable this functionality (e.g. to prevent failing tests on CI which uses production environment)
````
I'm missing info about native installation. eg. that you need to add new environment variable and it should be OK

s3tezsky replied:
What do you meaan by the `native instalation`? What you describe seems to mi like the second part of sentence right below.

s3tezsky commented on packages/framework/src/Model/Security/Authenticator.php:
```diff
@@ -52,7 +53,14 @@ public function checkLoginProcess(Request $request)
+            if ($error instanceof LoginWithDefaultPasswordException) {
+                $message = 'Oh, you just tried to log in using default credentials. We do not allow that on production'
+                    . ' environment. If you are random hacker, please go somewhere else. If you are authorized user,'
+                    . ' please use another account or contact developers and change password during deployment.';
```
Agree, but we should use plural in administrator too. I will fix it.

s3tezsky replied:
```suggestion
                    . ' please use another account or contact your developers or administrators and change password during deployment.';
```

s3tezsky commented on project-base/kubernetes/kustomize/overlays/ci/kustomization.yaml:
```diff
@@ -14,6 +14,12 @@ patchesJson6902:
         path: ./ingress-patch.yaml
+    -   target:
+            group: apps
+            version: v1
```
IMO it is. This patch is applied only on CI servers, where we do not want to disalow signing in witch default password.

s3tezsky commented:
@henzigo We added an environmet variable `IGNORE_DEFAULT_ADMIN_PASSWORD_CHECK` which might be set when testing on CI. By default we added it also into kubenetes settings. Otherwise you need to set it by yourself (by `Dockerfile`, `docker-composer.yml`, or some other way).

Is it OK for you now?

DavidOstrozlik commented on docs/introduction/running-acceptance-tests.md:
```diff
@@ -1,5 +1,7 @@
 # Running Acceptance Tests
 
+*Note: You may want to test administrator's interaction in production environment. In that case you should follow [Upgrade Notes](/docs/upgrade/UPGRADE-v8.1.0-dev.md#infrastructure) (search for `IGNORE_DEFAULT_ADMIN_PASSWORD_CHECK`) or change administrator's password since we do not allow administrators to log in with default credentials.*
```
```suggestion
*Note: You may want to test administrator's interaction in production environment. In that case you should follow [Upgrade Notes](/docs/upgrade/UPGRADE-v8.1.0-dev.md#infrastructure) (search for `IGNORE_DEFAULT_ADMIN_PASSWORD_CHECK`) or change administrator's password since we do not allow administrators to log in with default credentials in production environment.*
```

henzigo commented:
Hello, I just missing information about this feature in the documentation.

henzigo commented on packages/framework/src/Resources/config/parameters_common.yml:
```diff
@@ -13,3 +13,5 @@ parameters:
     # Default extended classes mapping
     shopsys.entity_extension.map: {}
+
+    env(IGNORE_DEFAULT_ADMIN_PASSWORD_CHECK): 0
```
Just question. Why is there 0/1 instead of false/true?

s3tezsky replied:
There is no reason for that.

PetrHeinz replied:
Changed to `'0'` in #1386 for consistency with env vars in general (they always contain strings)

LukasHeinz commented:
For documentation I would suggest FAQ and instruction for setting up a CI. I do not know better place right now.

s3tezsky commented:
The main goal of making this work around was to prevent failing acceptance tests. This is now OK, it is described in _upgrade note_ and also in _running acceptance tests_. Is it really necessary to add it into FAQ since no one haven't asked yet?

PetrHeinz commented:
@s3tezsky: I'm confused, acceptance tests use test environment, the feature shoudn't affect them at all...

s3tezsky commented:
@PetrHeinz Sorry about that there must be some missunderstanding from my side. The change will affects smoke tests.

Now I'm totally confused how to handle info for developers about ingoring the check 😕

henzigo commented:
smoke tests are using different way. I think that they are running with --env=test that is allowed in console.

henzigo commented:
or there is hard-coded way for using test environment

s3tezsky commented:
@henzigo Smoke tests are running in production environment. Check #1380 where our smoke tests started failing. But maybe it is somehow turned into prod. env. in backround.

PetrHeinz commented on packages/framework/src/Model/Security/AdministratorChecker.php:
```diff
@@ -0,0 +1,42 @@
+     */
+    public function checkPreAuth(UserInterface $user)
+    {
+        if (Environment::getEnvironment(false) === EnvironmentType::PRODUCTION
```
Resolved in #1386
